lib/init.coffee

```diff
@@ -102,23 +108,29 @@ module.exports =
       realpath = fs.realpathSync(p);
       return textEditor.getPath().substr(0, realpath.length) == realpath;
 
-  getFilesEndingWith: (startPath, endsWith) ->
+  getFilesEndingWith: (startPath, endsWith, ignoreFn) ->
     foundFiles = []
-    if !fs.existsSync(startPath)
-      return foundFiles
-    files = fs.readdirSync(startPath)
-    i = 0
-    while i < files.length
-      filename = path.join(startPath, files[i])
-      stat = fs.lstatSync(filename)
-      if stat.isDirectory()
-        foundFiles.push.apply(foundFiles, @getFilesEndingWith(filename, endsWith))
-      else if filename.indexOf(endsWith, filename.length - (endsWith.length)) >= 0
-        foundFiles.push.apply(foundFiles, [filename])
-        #Array::push.apply foundFiles, filename
-      i++
-    return foundFiles
-
+    folderFiles = []
+    voucher fs.readdir, startPath
+      .then (files) =>
+        folderFiles = files
+        Promise.all files.map (f) =>
```

Comment:
While not required, since I don't use `this` here, why not have it? Makes it closer to the ES6 spec, also consistenty.